xarray/tests/test_backends.py

```diff
@@ -468,6 +468,18 @@ def test_array_type_after_indexing(self):
             assert_identical(expected, actual)
             self.validate_array_type(actual)
 
+    def test_dropna(self):
+        # regression test for GH:issue:1694
+        a = np.random.randn(4, 3)
+        a[1, 1] = np.NaN
+        in_memory = xr.Dataset({'a': (('y', 'x'), a)},
+                               coords={'y': np.arange(4), 'x': np.arange(3)})
+        with self.roundtrip(in_memory) as on_disk:
+            self.validate_array_type(on_disk)
+            expected = in_memory.dropna(dim='x')
+            actual = on_disk.dropna(dim='x')
```

Comment:
You may want to add an assert that it actually drops the `NaN` in case this doesn't behave well for both in-memory and on-disk (though less important if this is tested elsewhere)